Plugin Vault/src/main/java/net/simon987/vaultplugin/VaultDoor.java

```diff
@@ -19,77 +20,148 @@
+    private int openedTimer = 0;
 
+    private int password_length;
+    private int cypher_id;
 
-    @Override
-    public void update() {
+    public VaultDoor(int password_length, int cypher_id){
+        this.password_length = password_length;
+        this.cypher_id = cypher_id;
+        this.random_string_generator = new RandomString(password_length);
 
-        if (openedTimer <= 0) {
+        password = getRandomPassword();
+    }
 
-            //Door was opened for OPEN_TIME, close it and regen password
-            password = getRandomPassword();
-            opened = false;
 
-            LogManager.LOGGER.fine("Closed Vault door ID: " + getObjectId());
-        } else {
-            openedTimer--;
+    @Override
+    public void update() {
+        if (open){
+            if (openedTimer <= 0) {
+                //Door was open for OPEN_TIME, close it and regen password
+                password = getRandomPassword();
+                open = false;
+                openedTimer = 0;
+                LogManager.LOGGER.fine("Closed Vault door ID: " + getObjectId());
+            } 
+            else{
+                openedTimer--;
+            }
         }
 
     }
 
     @Override
     public boolean sendMessage(char[] message) {
 
-        System.out.println("VAULT: sendMessage" + new String(message));//todo rmv
-
-        if (!opened) {
-
-            if (Arrays.equals(message, password)) {
-                opened = true;
-                openedTimer = OPEN_TIME;
-
-                LogManager.LOGGER.fine("Opened Vault door ID: " + getObjectId());
+        if (Arrays.equals(message, password)) {
+            if (!open) {
+                openVault();
+            } 
+            else{
+                keepVaultOpen();
             }
-
             return true;
-        } else {
-            //Can't receive messages when opened
+        }
+        else {
             return false;
         }
     }
 
+    private void openVault(){
+        open = true;
+        openedTimer = OPEN_TIME;
+        LogManager.LOGGER.fine("Opened Vault door ID: " + getObjectId());
+    }
+
+    private void keepVaultOpen(){
+        open = true;
+        openedTimer = OPEN_TIME;
+    }  
+
     @Override
     public boolean enter(GameObject object) {
 
-        LogManager.LOGGER.fine("VAULT enter " + opened);
-
-        if (opened) {
+        LogManager.LOGGER.fine("VAULT enter " + open);
 
+        if (open) {
             //TODO: Enter in the vault
-
-
             return true;
         } else {
             return false;
         }
-
     }
 
+    /**
+     * Generates a random alphanumeric string using the RandomString class
+     */
     private static char[] getRandomPassword() {
-        return "12345678".toCharArray();//todo actual random password
+        return random_string_generator.nextString().toCharArray();
+    }
+
+
+
+    private char[] encryptVernam(char[] plaintext){
```

Comment:
Should be in some util class since Radio Operator NPC will also use these functions (See [My comment](https://github.com/simon987/Much-Assembly-Required/issues/39#issuecomment-356125003) )